sdk/keyvault/azkeys/testdata/perf/unwrap.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/perf"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys/crypto"
+)
+
+type unwrapTestOptions struct{}
+
+var unwrapTestOpts unwrapTestOptions = unwrapTestOptions{}
+
+type unwrapTest struct {
+	perf.PerfTestOptions
+	keyName      string
+	client       *azkeys.Client
+	cryptoClient *crypto.Client
+	alg          crypto.WrapAlg
+	encryptedKey []byte
+}
+
+// newUnwrapTest is called once per process
+func newUnwrapTest(ctx context.Context, options perf.PerfTestOptions) (perf.GlobalPerfTest, error) {
+	d := &unwrapTest{
+		PerfTestOptions: options,
+		keyName:         "livekvtestunwrapperfkey",
+		alg:             crypto.WrapAlgRSAOAEP256,
+	}
+
+	vaultURL, ok := os.LookupEnv("AZURE_KEYVAULT_URL")
+	if !ok {
+		return nil, fmt.Errorf("the environment variable 'AZURE_KEYVAULT_URL' could not be found")
+	}
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	client, err := azkeys.NewClient(vaultURL, cred, &azkeys.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: options.Transporter,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.CreateRSAKey(ctx, d.keyName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	cryptoClient, err := crypto.NewClient(*resp.ID, cred, &crypto.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: options.Transporter,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	b := make([]byte, 32)
+	_, err = rand.Read(b)
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := cryptoClient.WrapKey(ctx, d.alg, b, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
As with decrypt, I'd move this to per-run to avoid caching of results: https://github.com/Azure/azure-sdk-for-net/blob/1b56a4b5c85e427d912dcc2ffa5f5a627bf3e5b7/sdk/keyvault/Azure.Security.KeyVault.Keys/perf/Scenarios/Unwrap.cs#L26-L36

Reply:
This runs once per-run right now